src/main/java/org/jline/utils/AttributedCharSequence.java

```diff
@@ -121,6 +122,13 @@ private static boolean attr(StringBuilder sb, String s, boolean first) {
 
     public abstract AttributedStyle styleAt(int index);
 
+    public int styleCodeAt(int index) {
```

Comment:
I think this should be protected or package level, given the values of the flags are not public.